packages/echo-core/src/services/baseClient/baseClient.ts

```diff
@@ -83,10 +83,11 @@ export default class BaseClient {
     ): Promise<Response> {
         let statusCode = 0;
         try {
+            const initialContentType = { 'Content-Type': 'application/json' };
```

Comment:
Yes, your are on the correct track. But this functionality is hidden magic from the outside. I think this can be solved by just moving it up as a default parameter. Then it's visible to the outside that this is default, and will be overwritten if they specify anything else.

instead of this:
```headerOptions: Record<string, unknown> = {},```
use this: 
```headerOptions: Record<string, unknown> = { 'Content-Type': 'application/json' },```